### PR TITLE
[kubeadm]: CoreDNS preflight: Parse correct version syntax

### LINKED ISSUE #2371: CoreDNS will not be upgraded: cannot migrate up to 'v1.8.0' from '1.7.0'
after merging the PR to update coredns to v1.8.0 in 1.21 our CI cough a problem:
https://github.com/kubernetes/kubernetes/pull/96429

https://k8s-testgrid.appspot.com/sig-cluster-lifecycle-kubeadm#kubeadm-kinder-upgrade-1-20-latest
https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-e2e-kubeadm-kinder-upgrade-1-20-latest/1344068108867866624/artifacts/task-06-upgrade-log.txt

```
[upgrade/config] FYI: You can look at this config file with 'kubectl -n kube-system get cm kubeadm-config -o yaml'
I1230 00:04:30.477142   13764 round_trippers.go:445] GET https://172.17.0.7:6443/api/v1/namespaces/kube-system/configmaps/kubeadm-config?timeout=10s 200 OK in 4 milliseconds
I1230 00:04:30.484415   13764 round_trippers.go:445] GET https://172.17.0.7:6443/api/v1/namespaces/kube-system/configmaps/kube-proxy?timeout=10s 200 OK in 4 milliseconds
I1230 00:04:30.495513   13764 round_trippers.go:445] GET https://172.17.0.7:6443/api/v1/namespaces/kube-system/configmaps/kubelet-config-1.20?timeout=10s 200 OK in 4 milliseconds
I1230 00:04:30.497557   13764 loader.go:372] Config loaded from file:  /etc/kubernetes/kubelet.conf
I1230 00:04:30.502737   13764 round_trippers.go:445] GET https://172.17.0.7:6443/api/v1/nodes/kinder-upgrade-control-plane-1?timeout=10s 200 OK in 4 milliseconds
I1230 00:04:30.516566   13764 round_trippers.go:445] GET https://172.17.0.7:6443/api/v1/namespaces/kube-system/pods?fieldSelector=spec.nodeName%3Dkinder-upgrade-control-plane-1&labelSelector=component%3Dkube-apiserver%2Ctier%3Dcontrol-plane 200 OK in 8 milliseconds
I1230 00:04:30.809228   13764 common.go:168] running preflight checks
I1230 00:04:30.809283   13764 preflight.go:87] validating if there are any unsupported CoreDNS plugins in the Corefile
[preflight] Running pre-flight checks.
I1230 00:04:30.814650   13764 round_trippers.go:445] GET https://172.17.0.7:6443/api/v1/namespaces/kube-system/configmaps/coredns?timeout=10s 200 OK in 5 milliseconds
I1230 00:04:30.819979   13764 round_trippers.go:445] GET https://172.17.0.7:6443/apis/apps/v1/namespaces/kube-system/deployments?labelSelector=k8s-app%3Dkube-dns 200 OK in 4 milliseconds
I1230 00:04:30.831820   13764 preflight.go:113] validating if migration can be done for the current CoreDNS release.
I1230 00:04:30.837379   13764 round_trippers.go:445] GET https://172.17.0.7:6443/api/v1/namespaces/kube-system/configmaps/coredns?timeout=10s 200 OK in 5 milliseconds
I1230 00:04:30.842846   13764 round_trippers.go:445] GET https://172.17.0.7:6443/apis/apps/v1/namespaces/kube-system/deployments?labelSelector=k8s-app%3Dkube-dns 200 OK in 5 milliseconds
I1230 00:04:30.847559   13764 round_trippers.go:445] GET https://172.17.0.7:6443/api/v1/namespaces/kube-system/configmaps/kube-dns?timeout=10s 404 Not Found in 4 milliseconds
[preflight] Some fatal errors occurred:
	[ERROR CoreDNSMigration]: CoreDNS will not be upgraded: cannot migrate up to 'v1.8.0' from '1.7.0'
[preflight] If you know what you are doing, you can make a check non-fatal with `--ignore-preflight-errors=...`
```

the problem can be reproduced with `kubeadm upgrade apply` with a kubeadm build from master.

### PR DESCRIPTION
**Pull request by rajansandeep**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:
Corrects the error during preflight: "CoreDNS will not be upgraded: cannot migrate up to 'v1.8.0' from '1.7.0'"

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes/kubeadm/issues/2371

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by neolit123** [comment]
thanks for the quick fix @rajansandeep 

/priority important-soon
/triage accepted
/lgtm
/approve

**Comment by rajansandeep** [comment]
/lgtm cancel 
@neolit123 I won’t be able to update the PR immediately (I’m on vacation with poor connectivity)

**Comment by hasheddan** [comment]
@neolit123 @rajansandeep mind if I open a PR to supersede this one to get https://testgrid.k8s.io/sig-release-master-informing#kubeadm-kinder-upgrade-1-20-latest healthy again?

**Comment by neolit123** [comment]
@hasheddan SGTM, thanks.

**Comment by rajansandeep** [comment]
Sorry for the delay @hasheddan @neolit123, I have updated this PR

**Review by hasheddan — APPROVED** [review]
Looks like @rajansandeep beat me to it :)

/lgtm

**Review by pacoxu — APPROVED** [review]
/lgtm
need to solve the strings dependencies before merging

**Comment by neolit123** [comment]
this is ready to go?
/hold cancel

**Review by pacoxu — APPROVED** [review]

**Comment by pacoxu** [comment]
ready

**Pull request merged**
